stock_picking_invoicing/views/stock_picking.xml

```diff
@@ -0,0 +1,80 @@
+        <field name="arch" type="xml">
+            <field name="product_id" position="before">
+                <filter name="picking_to_invoice" string="Shipments to Invoice" domain="[('invoice_state', '=', '2binvoiced')]"/>
+            </field>
+        </field>
+    </record>
+
+    <record id="view_picking_form" model="ir.ui.view">
+        <field name="name">stock.picking.form.inherit</field>
+        <field name="model">stock.picking</field>
+        <field name="inherit_id" ref="stock_picking_invoice_link.view_picking_form"/>
+        <field name="arch" type="xml">
+            <!-- The replace is needed because the tree view definition below change
+            also the field view defined in stock_picking_invoice_link -->
+            <field name="invoice_ids" position="replace"/>
+
+            <notebook position="inside">
+                <page string="Invoicing" name="page_invoicing" groups="account.group_account_invoice" attrs="{'invisible': [('invoice_state', 'in', [False, 'none'])]}">
+                    <group colspan="2">
+                        <field name="invoice_ids" coslpan="2" nolabel="1">
+                            <tree>
+                                <field name="partner_id"/>
+                                <field name="name"/>
+                                <field name="number"/>
+                                <field name="date_invoice"/>
+                                <field name="state"/>
+                                <field name="amount_untaxed" sum="True"/>
+                                <field name="amount_tax"/>
+                                <field name="amount_total" sum="True"/>
+                            </tree>
+                        </field>
+                    </group>
+                </page>
+            </notebook>
+            <button name="do_unreserve" position="after">
+                <button name="%(action_stock_invoice_onshipping)d" string="Create Invoice" attrs="{'invisible': ['|', ('state', '!=', 'done'), ('invoice_state', '!=', '2binvoiced')]}" type="action" class="oe_highlight" groups="base.group_user"/>
+            </button>
+            <field name="move_ids_without_package" position="attributes">
+                <attribute name="context">
+                    {'default_invoice_state': invoice_state,
+                    'address_in_id': partner_id,
+                    'form_view_ref':'stock.view_move_picking_form',
+                    'tree_view_ref':'stock.view_move_picking_tree',
+                    'default_picking_type_id': picking_type_id,
+                    'default_picking_id': id,
+                    'default_location_id': location_id,
+                    'default_location_dest_id': location_dest_id}
+                </attribute>
+            </field>
+            <field name="move_type" position="after">
+                <field name="invoice_state" groups="account.group_account_invoice"/>
```

Comment:
1. Can you put this field above the picking type?
2. You can't change the state of this field if the picking is not draft;
3. You must ensure the sync between this field and the stock.moves;
4. Should it be allowed some itens that will be invoiced and others not in the same picking?

**That's why I detected the bug of having an invoice without lines;**